File: maven-root/mtsa/src/main/java/RobotProtocol/src/robotProtocol/messages/CommandAcceptedMessage.cpp
/*
 * CommandAcceptedMessage.cpp
 *
 *  Created on: Feb 3, 2013
 *      Author: mariano
 */

#include "CommandAcceptedMessage.h"
#include <stdlib.h>
#include <stdio.h>

namespace RobotProtocol {

CommandAcceptedMessage CommandAcceptedMessage::staticMessage = CommandAcceptedMessage(0,0,0);

Message *CommandAcceptedMessage::deserialize(char *serializedMessage){
	CommandAcceptedMessage::staticMessage.initialize(serializedMessage);
	return &(CommandAcceptedMessage::staticMessage);
}

CommandAcceptedMessage::CommandAcceptedMessage(int to, int from, int acceptedMid)
:Message(MESSAGE_ID, 1, to, from){
	this->acceptedMid	= acceptedMid;
}

CommandAcceptedMessage::CommandAcceptedMessage(char *serializedMessage){
	initialize(serializedMessage);
}

CommandAcceptedMessage::~CommandAcceptedMessage() {
}

int CommandAcceptedMessage::getMessageLength(){
	return Message::getMessageLength() + ACCEPTED_MID_FIELD_LENGTH;
}

int CommandAcceptedMessage::getAcceptedMid(){
	return acceptedMid;
}


void CommandAcceptedMessage::readData(char **serializedMessageIndex){
	char *serializedMessage	= *serializedMessageIndex;
	char midField[ACCEPTED_MID_FIELD_LENGTH + 1];
	midField[ACCEPTED_MID_FIELD_LENGTH]	= '\0';
	int index = 0;
	memcpy(midField, &(serializedMessage[index]), ACCEPTED_MID_FIELD_LENGTH);
	acceptedMid = (int)strtol(midField, NULL, 10);
	index 		+= ACCEPTED_MID_FIELD_LENGTH;
	(*serializedMessageIndex) += index;
}

void CommandAcceptedMessage::writeData(char **serializedMessageIndex){
	sprintf(*serializedMessageIndex,"%04d", acceptedMid);
	(*serializedMessageIndex) = (*serializedMessageIndex) + ACCEPTED_MID_FIELD_LENGTH;
}

} /* namespace RobotProtocol */

File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/CommandAcceptedMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class CommandAcceptedMessage extends Message {

	public static int MESSAGE_ID				= 5;

	public static int ACCEPTED_MID_FIELD_LENGTH	= 4;	
	
	private int acceptedMid;
	
	public int getAcceptedMid(){
		return acceptedMid;
	}
	
	public CommandAcceptedMessage(int to, int from, int acceptedMid){
		super(MESSAGE_ID, 1, to, from);
		this.acceptedMid	= acceptedMid;
	}
	
	public CommandAcceptedMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength() + ACCEPTED_MID_FIELD_LENGTH;
	}
	
	
	@Override
	protected void readData(String inputString){
		int startIndex		= 0;
		int endIndex		= ACCEPTED_MID_FIELD_LENGTH;
		acceptedMid			= Integer.parseInt(inputString.substring(startIndex, endIndex),10);
	}

	@Override
	protected String writeData(){
		return String.format("%04d", acceptedMid);
	}

	@Override
	protected boolean hasReply() {
		return false;
	}

}
